### PR TITLE
General cleanup, warning suppression

### PR DESCRIPTION
**Pull request by JMY1000**
Wasn't sure if you'd want this, but I figured I might as well, so here it is.

Add documentation for some missing args, some more type inference, lots of suppressing warnings.

### CODE REVIEW

**Comment by JMY1000** [comment]
Apologies for the force pushes, screwed up the author and had to fix it.

File: src/main/java/uk/co/caprica/picam/Camera.java [hunk 1]
```diff
@@ -126,6 +126,7 @@ public boolean open() {
      * @return picture capture handler result
      * @throws CaptureFailedException if an error occurs
      */
+    @SuppressWarnings("UnusedReturnValue")
```
Comment (caprica) [inline src/main/java/uk/co/caprica/picam/Camera.java]: Why? The return value has a purpose. Presumably your application doesn't use the return value, but that doesn't mean it should be suppressed for everyone does it?

File: src/main/java/uk/co/caprica/picam/CameraConfiguration.java [hunk 2]
```diff
@@ -28,15 +28,14 @@
...
-import java.awt.geom.Rectangle2D;
-
 import static uk.co.caprica.picam.enums.Encoding.PNG;
 
 /**
  * Camera configuration.
  * <p>
  * This can be used in like a "builder".
  */
+@SuppressWarnings({"unused", "FieldCanBeLocal", "WeakerAccess"})
```
Comment (caprica) [inline src/main/java/uk/co/caprica/picam/CameraConfiguration.java]: I'm not sure Intellij specific inspections belong in a library. Also, there needs to be some justification/explanation for these - e.g. what access can be weaker here, what field can be local?

File: src/main/java/uk/co/caprica/picam/CameraConfiguration.java [hunk 3]
```diff
@@ -288,7 +287,7 @@ public CameraConfiguration captureTimeout(Integer captureTimeout) {
     /**
      * Create a new camera based on the configuration.
      *
-     * @return
+     * @return The constructed camera.
```
Comment (caprica) [inline src/main/java/uk/co/caprica/picam/CameraConfiguration.java]: It's not common convention to capitalise sentences nor use full-stop in Javadoc param or return annotations IMO

File: src/main/java/uk/co/caprica/picam/CaptureFailedException.java [hunk 4]
```diff
@@ -22,6 +22,7 @@
 /**
  * Exception thrown if a capture was requested but failed for some reason.
  */
+@SuppressWarnings({"unused", "WeakerAccess"})
```
Comment (caprica) [inline src/main/java/uk/co/caprica/picam/CaptureFailedException.java]: Same as before, what is it that is unused and what is it that can be weaker access?

File: src/main/java/uk/co/caprica/picam/PicamNativeLibrary.java [hunk 5]
```diff
@@ -47,6 +47,7 @@
  * PicamNativeLibrary.installLibrary("/home/pi");
  * </pre>
  */
+@SuppressWarnings({"unused", "WeakerAccess"})
```
Comment (caprica) [inline src/main/java/uk/co/caprica/picam/PicamNativeLibrary.java]: Same comment as before.

File: src/main/java/uk/co/caprica/picam/PicamNativeLibrary.java [hunk 6]
```diff
@@ -165,7 +166,7 @@ public static Path installLibrary(Path installDirectory, boolean overwrite) thro
                 if ("file".equalsIgnoreCase(protocol)) {
                     return installLibraryFromPath(Paths.get(containerUrl.toURI()), installDirectory, overwrite);
                 } else if ("jar".equalsIgnoreCase(protocol)) {
-                    try (FileSystem fs = FileSystems.newFileSystem(containerUrl.toURI(), Collections.<String, Object>emptyMap())) {
+                    try (FileSystem fs = FileSystems.newFileSystem(containerUrl.toURI(), Collections.emptyMap())) {
```
Comment (caprica) [inline src/main/java/uk/co/caprica/picam/PicamNativeLibrary.java]: Probably OK but depends on JDK/language level whether that inference can be made doesn't it?

File: src/main/java/uk/co/caprica/picam/PictureCaptureHandler.java [hunk 7]
```diff
@@ -24,6 +24,7 @@
  *
  * @param <T> type of result produced by the picture capture handler
  */
+@SuppressWarnings("unused")
```
Comment (caprica) [inline src/main/java/uk/co/caprica/picam/PictureCaptureHandler.java]: Same as before, what is the justification for this annotation?

File: src/main/java/uk/co/caprica/picam/SequentialFilePictureCaptureHandler.java [hunk 8]
```diff
@@ -27,6 +27,7 @@
  * A simple picture capture handler implementation that saves images to disk with sequentially
  * numbered filenames.
  */
+@SuppressWarnings("WeakerAccess")
```
Comment (caprica) [inline src/main/java/uk/co/caprica/picam/SequentialFilePictureCaptureHandler.java]: What can be weaker?

**Comment by JMY1000** [comment]
The idea was just to suppress warnings that occur when viewing the library as its own project. Since there are lots of parts of the code that aren't used **within** the library in the same way that they might be used when the library is being used as indented (as a library within another project), a user's development environment might issues warnings that don't hold any bearing.

To put it another way, the warning suppression isn't to suppress warnings that might pop up when using the library, but when developing the library itself.

**Pull request closed without merging**